src/Silex/Application.php

```diff
@@ -64,8 +63,12 @@ public function __construct(array $values = array())
 
         $app = $this;
 
-        $this['routes'] = function () {
-            return new RouteCollection();
+        $this['routes_class'] = 'Symfony\\Component\\Routing\\RouteCollection';
```

Comment:
I would remove this as overriding the `routes_factory` would do the same. So, it's not really a good extension point.